### PR TITLE
Docs/fxparams update modes and fx on

### PR DESCRIPTION
**Pull request by maerzhase**
This PR adds a first version of the documentation for fx(params) update modes and `$fx.on`

- https://fxhash-git-docs-fxparams-update-modes-and-fx-on-fxhash.vercel.app/doc/artist/params#fxparams-update-modes
- https://fxhash-git-docs-fxparams-update-modes-and-fx-on-fxhash.vercel.app/doc/artist/snippet-api#fxoneventid-handler-ondone

### CODE REVIEW

File: src/doc/artist/params.md [hunk 1]
```diff
@@ -38,6 +38,23 @@ $fx.params([
 $fx.getParam("number_id")
 ```
 
+# fx(params) update modes
+
+For each individual parameter that you specify, you can also set a specific update mode for. The default update mode is `"page-reload"` and describes the known behaviour of updating the artwork by performing a full page-load on the artwork. By setting the update mode to `"sync"` the parameter will be updated through the controllers without performaing a full page-load on the artwork.
```
Comment (louisholley) [inline src/doc/artist/params.md:43]:
```suggestion
For each individual parameter that you specify, you can also set a specific update mode for. The default update mode is `"page-reload"` and describes the known behaviour of updating the artwork by performing a full page-load on the artwork. By setting the update mode to `"sync"` the parameter will be updated through the controllers without performing a full page-load on the artwork.
```

File: src/doc/artist/params.md [hunk 2]
```diff
@@ -38,6 +38,23 @@ $fx.params([
...
+  id: "number_id",
+  name: "A number/float64",
+  type: "number",
+  update: "sync", // <-- new update property
+},
+```
+
+All updates on prameters with `update: "sync"` are received in the background in real-time. If your artwork is already running in some kind of draw loop (e.g. via `requestAnimationFrame`), you will receive those changes during the runtime of your loop.
+
+For more even more advanced usecases, e.g. if you want to only re-render you artwork when the values of your parameters are changing, you can use the new event listeners described in the section [`$fx.on`](/doc/artist/snippet-api#fxoneventid-handler-ondone) on the snippet api page.
```
Comment (louisholley) [inline src/doc/artist/params.md:56]:
```suggestion
For even more advanced use cases, e.g. if you want to only re-render your artwork when the values of your parameters are changing, you can use the new event listeners described in the section [`$fx.on`](/doc/artist/snippet-api#fxoneventid-handler-ondone) on the snippet api page. 
```

File: src/doc/artist/snippet-api.md [hunk 3]
```diff
@@ -531,6 +581,13 @@ A default value for the parameter. If defined, when the minting interface with t
 
 ---
 
+`update`
+
+_Optional_
+
+Specifies the update mode of the parameter. The default update mode is `"page-reload"`. With update mode `"page-reload"`, a full page reload on the artwork is performed whenever the value of the parameter changes. When the update mode is set to `"sync"` the values of the parameter are updated during the runtime of the artwork. With update mode `"sync"`, no page reload is performed when the the values of the parameter changed.
```
Comment (louisholley) [inline src/doc/artist/snippet-api.md:588]:
```suggestion
Specifies the update mode of the parameter. The default update mode is `"page-reload"`. With update mode `"page-reload"`, a full page reload on the artwork is performed whenever the value of the parameter changes. When the update mode is set to `"sync"` the parameter values are updated during the runtime of the artwork - no page reload is performed when the parameter values change.
```

**Review by louisholley — APPROVED** [review]
noice!

File: src/doc/artist/params.md [hunk 4]
```diff
@@ -38,6 +38,23 @@ $fx.params([
 $fx.getParam("number_id")
 ```
 
+# fx(params) update modes
+
+For each individual parameter that you specify, you can also set a specific update mode for. The default update mode is `"page-reload"` and describes the known behaviour of updating the artwork by performing a full page-load on the artwork. By setting the update mode to `"sync"` the parameter will be updated through the controllers without performing a full page-load on the artwork.
```
Comment (ciphrd) [inline src/doc/artist/params.md:43]: `et a specific update mode for.` -> `for it`

**Review by ciphrd — CHANGES_REQUESTED** [review]

**Review by louisholley — APPROVED** [review]

**Pull request merged**